blockmanager.go

```diff
@@ -2357,9 +2356,11 @@ func (b *blockManager) SetParentTemplate(bt *BlockTemplate) {
 
 // newBlockManager returns a new Decred block manager.
 // Use Start to begin processing asynchronous block and inv updates.
-func newBlockManager(s *server, indexManager blockchain.IndexManager, interrupt <-chan struct{}) (*blockManager, error) {
+func newBlockManager(s *server, indexManager blockchain.IndexManager, chain *blockchain.BlockChain, txMemPool *mempool.TxPool, interrupt <-chan struct{}) (*blockManager, error) {
```

Comment:
No need to take `chain` and `txMemPool` as separate params here.  They're available on `s` to set into the `blockManager` struct below.

I realize the intention is to decouple the block manager from the server, and adding the `chain` and `txMemPool` instances on the `blockManager` type makes perfect sense.  However, a `blockManagerConfig` struct is going to be needed in a future commit anyway which will contain the instances to be set, so there is no reason to introduce extra churn in the caller for the intermediate commits.